### PR TITLE
Add content page current state context

### PR DESCRIPTION
**Pull request by aomi**
## Overview

Closes #XX

- Changes the button "Bookmark" to "Add to Timetable" and "Remove from Timetable" as bookmarking can be confusing on whether it's a browser bookmark. Could simplify the wording but for now it's much more clear to the user.
- Adds skeleton loading to the course content title / description and links. Before it allowed `undefined` links which would be broken. 
- Disables the add/delete button while data is loading. 
- Moves the CRN to the left given it's not very important.
- Course detail badges are smaller now.  
- Adds toast messages when course is added/deleted via the button on the course details view.
- When getting course details errors out, shows the error message (shown below)

image

### CODE REVIEW

File: src/app/content/components/Section.tsx [hunk 1]
```diff
@@ -72,42 +72,44 @@ export function SectionInfo({
...
+            <Popover>
+              <PopoverTrigger>
+                <Badge colorScheme="green" mx="1">
+                  {instructionalMethod}
+                </Badge>
+              </PopoverTrigger>
+              <PopoverContent>
+                <PopoverArrow />
+                <PopoverCloseButton />
+                <PopoverBody>This means this section takes place online.</PopoverBody>
```
Comment (keithradford) [inline src/app/content/components/Section.tsx:90]: I know this is unrelated to this PR, but if `instructionalMethod` is in-person, does the popover body still say `This means this section takes place online.`?

**Review by keithradford — APPROVED** [review]
LGTM, just a side note

File: src/app/content/components/Section.tsx (hunk 1 above)
Comment (aomi) [reply src/app/content/components/Section.tsx:90]: No :disappointed:
